File: src-tauri/src/tests/unit/history.rs
use crate::core::traits::HistoryRepository;
use crate::core::types::HistoryEntry;
use crate::infra::history::SqliteHistoryRepository;
use rusqlite::Connection;

#[test]
fn test_history_lifecycle() {
    let conn = Connection::open_in_memory().unwrap();
    let repo = SqliteHistoryRepository::new(conn);

    // Init DB
    assert!(repo.init().is_ok());

    // Save entry
    let entry = HistoryEntry {
        id: "h1".to_string(),
        service_id: Some("s1".to_string()),
        endpoint_id: Some("e1".to_string()),
        method: "GET".to_string(),
        url: "/test".to_string(),
        request_headers: vec![],
        request_body: "".to_string(),
        response_status: 200,
        response_status_text: "OK".to_string(),
        response_headers: vec![],
        response_body: "body".to_string(),
        time_elapsed: 10,
        size: 4,
        created_at: "2023-01-01T00:00:00Z".to_string(),
    };

    assert!(repo.save(entry.clone()).is_ok());

    // Get history
    let history = repo.get_history(10, 0).unwrap();
    assert_eq!(history.len(), 1);
    assert_eq!(history[0].id, "h1");

    // Clear history
    assert!(repo.clear().is_ok());
    let history = repo.get_history(10, 0).unwrap();
    assert_eq!(history.len(), 0);
}

File: internal/models/history.go
package models

type HistoryEntry struct {
	ID                 string   `json:"id" yaml:"id"`
	ServiceID          *string  `json:"serviceId,omitempty" yaml:"serviceId,omitempty"`
	EndpointID         *string  `json:"endpointId,omitempty" yaml:"endpointId,omitempty"`
	Method             string   `json:"method" yaml:"method"`
	URL                string   `json:"url" yaml:"url"`
	RequestHeaders     []Header `json:"requestHeaders" yaml:"requestHeaders"`
	RequestBody        string   `json:"requestBody" yaml:"requestBody"`
	ResponseStatus     uint16   `json:"responseStatus" yaml:"responseStatus"`
	ResponseStatusText string   `json:"responseStatusText" yaml:"responseStatusText"`
	ResponseHeaders    []Header `json:"responseHeaders" yaml:"responseHeaders"`
	ResponseBody       string   `json:"responseBody" yaml:"responseBody"`
	TimeElapsed        uint64   `json:"timeElapsed" yaml:"timeElapsed"`
	Size               uint64   `json:"size" yaml:"size"`
	CreatedAt          string   `json:"createdAt" yaml:"createdAt"`
}
